pyproject.toml

```diff
@@ -36,9 +36,9 @@ dev = [
     "pytest",
     "numpy",
     "pre-commit",
-    "black==24.3.0",
+    "black==24.8.0",
     "isort",
-    "mypy==0.991",
+    "mypy==1.13.0",
```

Comment:
Are these tools run OUTSIDE of pre-commit?

Reply:
Sometimes, yes, though not automatically. I certainly prefer to have them installed in my venv and ready to call from the bash prompt. In all my projects, `pip install -e .[dev]` will give me black, isort, mypy, usually flake8 too, with whatever configuration is needed to "just work".